pkgs/data/fonts/windows-fonts/default.nix

```diff
@@ -0,0 +1,97 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchurl,
+  p7zip,
+}:
+let
+  windows-fonts =
+    {
+      pname,
+      version,
+      url,
+      hash,
+      desc,
+    }:
+    let
+      languageFonts = import ./${pname}-languages.nix;
+      source =
+        with lib.attrsets;
+        mapAttrs' (
+          name: fonts:
+          nameValuePair (name) (builtins.concatStringsSep " " (map (font: "Windows/Fonts/" + font) fonts))
+        ) languageFonts;
+    in
+    stdenvNoCC.mkDerivation {
+      inherit pname version desc;
+
+      src = fetchurl {
+        inherit url hash;
+        meta.license = lib.licenses.unfree;
+      };
+
+      outputs = [ "out" ] ++ builtins.attrNames source;
+
+      nativeBuildInputs = [ p7zip ];
+
+      unpackPhase = ''
+        runHook preUnpack
+        7z x $src
+        7z x -aoa sources/install.wim Windows/{Fonts/"*".{ttf,ttc},System32/Licenses/neutral/"*"/"*"/license.rtf}
+        runHook postUnpack
+      '';
+
+      installPhase =
+        ''
+          runHook preInstall
+
+          mkdir -p $out/share/fonts/truetype
+        ''
+        + lib.strings.concatLines (
+          lib.lists.forEach (builtins.attrNames source) (
+            name:
+            let
+              outHome = "$" + name;
+            in
+            (''
+              install -Dm644 ${source.${name}} -t ${outHome}/share/fonts/truetype
+              ln -s ${outHome}/share/fonts/truetype/*.{ttf,ttc} $out/share/fonts/truetype
+              install -Dm644 Windows/System32/Licenses/neutral/*/*/license.rtf -t ${outHome}/share/licenses
+            '')
+          )
+        )
```

Comment:
I recognize that it is probably subjective which one is more hedious than the other, but I don't like this code generation at all. I would replace this with `__structuredAttrs = true;`. You can use an associative array to get output paths in that case. See another (rather, the only other?) example in Nixpkgs: https://github.com/NixOS/nixpkgs/blob/240db6aa316f1ba7ffcee7cff12de6838e577ec3/pkgs/tools/typesetting/tex/texlive/build-texlive-package.nix#L288-L294

However, the language list is still too complicated for bash associative arrays. I can see two options:

1. `mapAttrs` them into an attrset of space-separated file names, do word split in bash
2. invert the map, make them `{ "font.ttf" = "language"; ... }` instead